ldap/servers/slapd/log.c

```diff
@@ -4923,13 +4923,17 @@ log__fix_rotationinfof(char *pathname)
                     ignoreit = 1;
                 }
             }
-            if (ignoreit || (q - p != 15)) {
+            /*
+             * Accept timestamp length of 15 (YYYYMMDD-HHMMSS) for uncompressed,
+             * or 18 (YYYYMMDD-HHMMSS.gz) for compressed files.
+             */
+            if (ignoreit || ((q - p != 15) && (q - p != 18))) {
```

Comment:
Should also ignore if q-p == 18 && strcmp(p + 15, ".gz"))

Maybe we should rather simplify the code and use a regex ?